debian/cloudstack-agent.postinst

```diff
@@ -50,6 +50,13 @@ case "$1" in
             mkdir /etc/libvirt/hooks
         fi
         cp -a /usr/share/cloudstack-agent/lib/libvirtqemuhook /etc/libvirt/hooks/qemu
+
+        # Enable TLS enabled VM migration for libvirtd
+        if ! iptables-save | grep -- "-A INPUT -p tcp -m tcp --dport 16514 -j ACCEPT" > /dev/null; then
+            iptables -t filter -A INPUT -p tcp -m tcp --dport 16514 -j ACCEPT
+            iptables-save > /etc/iptables/rules.v4
+        fi
+
```

Comment:
I am not to keen on this one. Do we really want our packages to start configuring a firewall on a host?

This should be on the docs to tell people to open the port(s), not having packages doing it manually.

In addition, the directory /etc/iptables does not exist by default on Ubuntu.

Reply:
The `cloudstack-setup-agent` reconfigures libvirtd config and also adds iptables rules for several ports, the post-install script (both rpm+deb) does a test if expected iptables rules are in place and adds the ACCEPT rule only if needed.

Given not all users may use a config mgmt system such as chef/puppet/ansible, running the commands as part of post-install script will save the additional work they may need to do themselves (manually or automated). @wido I'm okay to advise users via release/admin docs, but I don't see any negative/side-effects with the change.

Reply:
If users are using firewalld or ufw on their CentOS/Ubuntu system this may break things. And like I said, /etc/iptables does not exist on Ubuntu systems by default, you need the iptables-persistent package for that.

I wouldn't touch the firewall in a postinst of a package. The package should not touch parts of the system it's not configuring.

Reply:
Alright @wido, perhaps we can remove this for Ubuntu (debian pkg). On both CentOS 6 and 7, iptables service is indeed available that is used to save existing rules, firewalld is not used here. It is likely that things may break for el6/7 users. I'm okay to document the change in release notes docs as well. Let's ask others for their thoughts - @DaanHoogland @rafaelweingartner @resmo @ustcweizhou @nvazquez @mlsorensen ?

Reply:
Is there an alternative that will work. I would not like to see a large devide between the amount of work to be done on rhel-like systems versus debian-likes.

Reply:
Okay, removed this.